sdk/identity/src/token_credentials/client_secret_credentials.rs

```diff
@@ -135,7 +135,7 @@ impl TokenCredential for ClientSecretCredential {
         let token_result = client
             .exchange_client_credentials()
             .add_scope(Scope::new(format!("{}/.default", resource)))
-            .request_async(async_http_client)
+            .request_async(crate::oauth2_async_http_client)
```

Comment:
If would be nice if all functions in this crate had access to a `&dyn azure_core::HttpClient` this would make this crate completely unaware of which implementation it's using.

Reply:
Done. Header conversion and testing is left if you with to pick it up.